flink-clients/src/main/java/org/apache/flink/client/program/Client.java

```diff
@@ -277,12 +278,20 @@ else if (prog.isUsingInteractiveMode()) {
 			// invoke here
 			try {
 				prog.invokeInteractiveModeForExecution();
+				// if we're here, it means the program is valid for detached mode execution.
+				JobWithJars plan = ContextEnvironment.ContextEnvironmentFactory.getPlan();
+				JobGraph graph = ContextEnvironment.ContextEnvironmentFactory.getGraph();
+				if (plan != null && graph == null) {
```

Comment:
Ah. I had not considered this possibility. What should be done in such a case? Execute both programs? That does make sense, but what should be the return value?